### DISCUSSION TITLE
Java version support requirements

### DISCUSSION THREAD

**Discussion by ivakoleva** [vmware/versatile-data-kit#2105] [General]
I noticed we do run tests in CI/CD using:
```
(32/49) Installing openjdk17-jre (17.0.6_p10-r0)
(33/49) Installing openjdk17-jmods (17.0.6_p10-r0)
(34/49) Installing openjdk17-jdk (17.0.6_p10-r0)
```
While our [build requirements](https://github.com/vmware/versatile-data-kit/tree/main/projects/control-service#build) are specifically towards Java 11:
```
Java 11 and Docker are required.
```

I tried compiling with Java 11, our dependencies are too ahead of time. Also, we have a `record` keyword of >=14 introduced in https://github.com/vmware/versatile-data-kit/pull/1787 in April. Therefore, at the moment, in fact we do not support Java 11 at all.

There are multiple mentions of Java 11+ requirement written, for example:
https://github.com/vmware/versatile-data-kit/blob/main/cicd/build.sh#L11

**We need to sync our Java versions requirements understanding.**

**Comment by murphp15** [comment]
We support java 17+. The docs are out of date.
The reason we support java 17+ is that we want to be able to migrate to the latest spring eventually https://spring.io/blog/2022/05/24/preparing-for-spring-boot-3-0

**Comment by ivakoleva** [comment]
Attaching another requirement of Java 11+:
https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/CONTRIBUTING.md#run-the-control-service-locally
and https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/CONTRIBUTING.md#run-the-control-service-using-helm-and-k8s
